### PR TITLE
lrbd to ceph-iscsi upgrade fixes

### PR DESCRIPTION
**Pull request by ddiss**
Fixes:
bsc#1145617 - TPG auth settings lost during upgrade
bsc#1145618 - TPG LUN id lost during upgrade

Description:
I've only tested these changes manually so far. For full coverage I'm working on a rapido based VM which steps through lrbd->ceph-iscsi upgrades for most existing lrbd sample configs.

-----------------

**Checklist:**
- [ ] Added unittests and or functional tests
- [ ] Adapted documentation
- [ x] Referenced issues or internal bugtracker
- [ ] Ran integration tests successfully (trigger with "@susebot run teuthology" in a GitHub comment; see the [wiki](https://github.com/SUSE/DeepSea/wiki/Testing) for more information)

### CODE REVIEW

**Comment by rjfd** [comment]
@ricardoasmarques this change depends on the patch you have for ceph-iscsi in https://github.com/ceph/ceph-iscsi/pull/133 , correct?

**Comment by ddiss** [comment]
> @ricardoasmarques this change depends on the patch you have for ceph-iscsi in [ceph/ceph-iscsi#133](https://github.com/ceph/ceph-iscsi/pull/133) , correct?

That's correct. We'll need to have an explicit ceph-iscsi version dependency check somewhere.

**Comment by jschmid1** [comment]
linter complaints:

```
15:11:58 E:500, 0: PEP8 E128: continuation line under-indented for visual indent (continuation-line-under-indented-for-visual-indent)
15:11:58 E:501, 0: PEP8 E122: continuation line missing indentation or outdented (continuation-line-missing-indentation-or-outdented)
15:11:58 E:507, 0: PEP8 E128: continuation line under-indented for visual indent (continuation-line-under-indented-for-visual-indent)
15:11:58 E:537, 0: PEP8 E128: continuation line under-indented for visual indent (continuation-line-under-indented-for-visual-indent)
15:11:58 E:538, 0: PEP8 E128: continuation line under-indented for visual indent (continuation-line-under-indented-for-visual-indent)
15:11:58 E:539, 0: PEP8 E128: continuation line under-indented for visual indent (continuation-line-under-indented-for-visual-indent)
15:11:58 E:540, 0: PEP8 E128: continuation line under-indented for visual indent (continuation-line-under-indented-for-visual-indent)
15:11:58 C:344, 0: Line too long (107/100) (line-too-long)

```

**Comment by jschmid1** [comment]
Linter issues are fixed, @ricardoasmarques and @rjfd would you mind reviewing it again?

**Comment by jschmid1** [comment]
> Linter issues are fixed, @ricardoasmarques and @rjfd would you mind reviewing it again?

bump

**Comment by ricardoasmarques** [comment]
@jschmid1  this PR looks good but we cannot approve/merge it until https://github.com/ceph/ceph-iscsi/pull/133 is approved/merged.

If the `config` structure proposed in https://github.com/ceph/ceph-iscsi/pull/133 change during the review process, this DeepSea PR has to be adapted accordingly.

**Comment by jschmid1** [comment]
> @jschmid1 this PR looks good but we cannot approve/merge it until [ceph/ceph-iscsi#133](https://github.com/ceph/ceph-iscsi/pull/133) is approved/merged.
> 
> If the `config` structure proposed in [ceph/ceph-iscsi#133](https://github.com/ceph/ceph-iscsi/pull/133) change during the review process, this DeepSea PR has to be adapted accordingly.

alright, do you mind keeping us posted @ricardoasmarques ?

**Comment by ricardoasmarques** [comment]
@jschmid1 Sure, I'll let you know as soon as I have news from the `ceph-iscsi` side.

**Comment by ddiss** [comment]
I have a follow-up here to further restrict mixing of TPG auth and ACLs, to go with Ricardo's reworked ceph-iscsi changes. Please don't merge this yet.

**Comment by ddiss** [comment]
> I have a follow-up here to further restrict mixing of TPG auth and ACLs, to go with Ricardo's reworked ceph-iscsi changes. Please don't merge this yet.

I've pushed the extra combined TPG + ACL auth check. @rjfd I'm still finishing off testing, but it'd be great if you could review these changes.

File: srv/salt/_modules/iscsi.py [hunk 1]
```diff
@@ -425,20 +456,60 @@ def validate(lio_root):
...
+            if tpg_auth_prev is None:
+                tpg_auth_prev = tpg_auth
+            elif tpg_auth_prev != tpg_auth:
+                raise Exception(
+                    'Unsupported LIO configuration: Authentication settings '
+                    'differ between TPGs for target ({}). Switch lrbd to target'
+                    'based authentication to proceed.'.format(target.wwn))
+            if tpg_acls_prev is None:
+                tpg_acls_prev = tpg.node_acls
+            elif tpg_acls_prev != tpg.node_acls:
```
Comment (rjfd) [inline srv/salt/_modules/iscsi.py]: `tpg.node_acls` is a list generator and cannot be use directly with a comparison operator like `==`.
Change this code to:
```python
    tpg_acls_prev = list(tpg.node_acls)
elif tpg_acls_prev != list(tpg.node_acls):
```

**Comment by rjfd** [comment]
I pushed a commit to fix the bug I commented above and the linting issues.
I've tested the upgrade and it is working fine.

**Comment by ricardoasmarques** [comment]
@jschmid1 we now have the issue number in the commit title, shouldn't we change the commit messages to include
`Fixes: https://bugzilla.suse.com/show_bug.cgi?id=1145617`
and 
`Fixes: https://bugzilla.suse.com/show_bug.cgi?id=1145618`? 

From the code changes point of view this PR is ready to be merged.

@jschmid1 as soon as this PR is merged @rjfd can take care of creating the backport.

**Comment by jschmid1** [comment]
> @jschmid1 we now have the issue number in the commit title, shouldn't we change the commit messages to include
> `Fixes: https://bugzilla.suse.com/show_bug.cgi?id=1145617`
> and
> `Fixes: https://bugzilla.suse.com/show_bug.cgi?id=1145618`?
> 
> From the code changes point of view this PR is ready to be merged.
> 
> @jschmid1 as soon as this PR is merged @rjfd can take care of creating the backport.

nah, that's all fine :+1:

**Pull request merged**